app/Jobs/DeleteGenre.php

```diff
@@ -67,14 +63,14 @@ public function handle()
 
         // This is done instead of a single UPDATE query in order to
         // generate revision logs for the change.
-        $this->genreToDelete->tracks()->chunk(200, function ($tracks) {
-            foreach ($tracks as $track) {
-                /** @var Track $track */
+        $this->genreToDelete->tracks()->each(function ($track) {
+            /** @var Track $track */
 
-                $track->genre_id = $this->destinationGenre->id;
-                $track->save();
-                $track->updateTags();
-            }
+            $track->genre_id = $this->destinationGenre->id;
+            $track->save();
+            $track->updateTags();
         });
+
+        $this->genreToDelete->delete();
```

Comment:
I'm concerned about the race condition that this change introduces - someone could link a track to a genre that's in the middle of being deleted. I suggest an alternate approach: querying the tracks to modify by `genre_id` rather than through the Genre object's `tracks` relation.